File: 2024-06-week4/2559.js
const fs = require("fs");
const io = process.platform === "linux" ? "/dev/stdin" : "input.txt";

let [meta, arr] = fs.readFileSync(io).toString().trim().split("\n");

let [n, k] = meta.split(" ").map(Number);
arr = arr.split(" ").map(Number);

let prefixSum = [0];

for (let i = 1; i <= n; i++) {
    prefixSum[i] = prefixSum[i - 1] + arr[i - 1];
}

let max = -Infinity;

for (let index = k; index <= n; index++) {
    let sum = prefixSum[index] - prefixSum[index - k];
    max = sum > max ? sum : max;
}
console.log(max);

File: 백준_파이썬/2559.py
N, L = map(int, input().split())

arr = list(map(int, input().split()))
prefix_sum = arr[:] + [0]

for i in range(N):
    prefix_sum[i] += prefix_sum[i-1]

max_sum = float("-inf")

for idx in range(N):
    if idx + L - 1 < N:
        max_sum = max(prefix_sum[idx + L - 1] - prefix_sum[idx - 1], max_sum)
    
print(max_sum)